data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java

```diff
@@ -0,0 +1,189 @@
+package com.microsoft.azure.kusto.data;
+
+import com.azure.core.http.*;
+import com.azure.core.util.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.microsoft.azure.kusto.data.exceptions.*;
+import com.microsoft.azure.kusto.data.http.CloseParentResourcesStream;
+import com.microsoft.azure.kusto.data.http.HttpRequestBuilder;
+import com.microsoft.azure.kusto.data.http.HttpStatus;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.conn.EofSensorInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.Optional;
+import java.util.zip.DeflaterInputStream;
+import java.util.zip.GZIPInputStream;
+
+public abstract class BaseClient implements Client, StreamingClient {
```

Comment:
This was mainly to get rid of Utils and store the underlying HTTPClient in the "Kusto Client" so that we could stop passing the HTTPClient around in the code so much and so often. I made the class abstract so that there could be multiple Client implementations over time. 

If there's no plan to have other implementations we could move the logic from this class into the ClientImpl. I am wondering if we will want an AsyncClient implementation at some point or just add methods with async (flux/mono) signatures to the existing client implementation. If you're preference is the former, my recommendation would be that we keep it, if it's the latter my recommendation would be to move the logic from this abstract class into ClientImpl, and get rid of this class.